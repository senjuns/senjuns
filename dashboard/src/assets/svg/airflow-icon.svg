<svg width="26" height="24" viewBox="0 0 26 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M25.8886 7.50005C25.8886 10.4777 23.4962 12.9 20.5553 12.9H0.999729C0.509062 12.9 0.11084 12.4968 0.11084 12C0.11084 11.5032 0.509062 11.1 0.999729 11.1H20.5553C22.5157 11.1 24.1108 9.485 24.1108 7.50005C24.1108 5.5151 22.5157 3.90005 20.5553 3.90005C18.5948 3.90005 16.9997 5.5151 16.9997 7.50005C16.9997 7.99685 16.6015 8.40005 16.1108 8.40005C15.6202 8.40005 15.2219 7.99685 15.2219 7.50005C15.2219 4.5224 17.6144 2.10005 20.5553 2.10005C23.4962 2.10005 25.8886 4.5224 25.8886 7.50005ZM0.999729 8.40005H9.88862C12.0944 8.40005 13.8886 6.58295 13.8886 4.35005C13.8886 2.11715 12.0944 0.300049 9.88862 0.300049C7.68284 0.300049 5.88862 2.11715 5.88862 4.35005C5.88862 4.84685 6.28684 5.25005 6.77751 5.25005C7.26817 5.25005 7.66639 4.84685 7.66639 4.35005C7.66639 3.1094 8.66328 2.10005 9.88862 2.10005C11.1139 2.10005 12.1108 3.1094 12.1108 4.35005C12.1108 5.5907 11.1139 6.60005 9.88862 6.60005H0.999729C0.509062 6.60005 0.11084 7.00325 0.11084 7.50005C0.11084 7.99685 0.509062 8.40005 0.999729 8.40005ZM16.1108 15.6H0.999729C0.509062 15.6 0.11084 16.0032 0.11084 16.5C0.11084 16.9968 0.509062 17.4 0.999729 17.4H16.1108C17.3362 17.4 18.3331 18.4094 18.3331 19.65C18.3331 20.8907 17.3362 21.9 16.1108 21.9C14.8855 21.9 13.8886 20.8907 13.8886 19.65C13.8886 19.1532 13.4904 18.75 12.9997 18.75C12.5091 18.75 12.1108 19.1532 12.1108 19.65C12.1108 21.8829 13.9051 23.7001 16.1108 23.7001C18.3166 23.7001 20.1108 21.8829 20.1108 19.65C20.1108 17.4172 18.3166 15.6 16.1108 15.6Z" fill="black"/>
</svg>
