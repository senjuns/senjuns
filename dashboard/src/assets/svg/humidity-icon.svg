<svg width="28" height="32" viewBox="0 0 28 32" fill="none" xmlns="http://www.w3.org/2000/svg">
  <g clip-path="url(#clip0_2365_9474)">
    <path d="M15.7557 8.84552C15.7557 5.50667 10.8455 0 10.8455 0C10.8455 0 5.9375 5.49646 5.9375 8.84552C5.9375 11.4034 7.50958 13.7565 10.8484 13.7565C14.1873 13.7565 15.7557 11.208 15.7557 8.84552ZM10.8346 11.7943C9.45792 11.7943 7.88583 11.2109 7.88583 8.84552C7.88583 6.88333 10.8346 3.14417 10.8346 3.14417C10.8346 3.14417 13.7833 6.87312 13.7833 8.84552C13.7833 11.208 12.2083 11.7935 10.8353 11.7935L10.8346 11.7943Z" fill="black"/>
    <path d="M10.4847 26.1788C10.4847 22.8399 5.57452 17.3333 5.57452 17.3333C5.57452 17.3333 0.666504 22.8297 0.666504 26.1788C0.666504 28.7367 2.23859 31.0897 5.57744 31.0897C8.91629 31.0897 10.4847 28.5413 10.4847 26.1788ZM5.56359 29.1275C4.18692 29.1275 2.61484 28.5442 2.61484 26.1788C2.61484 24.2166 5.56359 20.4774 5.56359 20.4774C5.56359 20.4774 8.51234 24.2064 8.51234 26.1788C8.51234 28.5413 6.93734 29.1268 5.56432 29.1268L5.56359 29.1275Z" fill="black"/>
    <path d="M21.5774 12C21.5774 12 16.6665 17.4965 16.6665 20.8455C16.6665 23.4034 18.2386 25.7565 21.5774 25.7565C24.9163 25.7565 26.4855 23.1985 26.4855 20.8455C26.4855 17.4965 21.5774 12 21.5774 12ZM21.5774 23.7833C20.2008 23.7833 18.6287 23.2 18.6287 20.8346C18.6287 18.8724 21.5774 15.1332 21.5774 15.1332C21.5774 15.1332 24.5262 18.8622 24.5262 20.8346C24.5262 23.1985 22.9541 23.7841 21.5774 23.7841V23.7833Z" fill="black"/>
  </g>
  <defs>
    <clipPath id="clip0_2365_9474">
      <rect width="26.6667" height="32" fill="white" transform="translate(0.666504)"/>
    </clipPath>
  </defs>
</svg>
