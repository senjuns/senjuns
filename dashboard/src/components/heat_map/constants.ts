// TODO - Update the resource path to real room image.
export const ROOM_IMAGE_PATH = '/zone_map.png';

// Colors that are being used to display the heatmap in gradient.
export const COLOR_SCALE_VALUES: Array<[number, string]> = [
  [0, '#0046B3'], // Blue
  [0.33, '#6FC153'], // Green
  [0.67, '#FFB94C'], // Yellow
  [1, '#E83339'], // Red
];

export const FRAME_TRANSITION_DURATION = 500;
