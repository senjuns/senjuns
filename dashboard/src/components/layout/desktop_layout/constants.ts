// Header height of the desktop layout
export const HEADER_HEIGHT = 110;

// Sidebar width of the desktop layout
export const SIDEBAR_WIDTH = 110;
