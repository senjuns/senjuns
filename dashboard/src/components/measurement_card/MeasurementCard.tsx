import React, { MouseEvent } from 'react';
import styled from 'styled-components';
import {
  Colors,
  NotificationLevel,
  Sizes,
  ScreenSize,
} from '../../shared/constants';

const colorMap = {
  [NotificationLevel.Nominal]: Colors.black,
  [NotificationLevel.Warning]: Colors.orange3,
  [NotificationLevel.Error]: Colors.orange1,
};

interface PropsType {
  icon: JSX.Element;
  title: string;
  value?: number;
  unit: string;
  notificationLevel: string;
  decimal?: number;
  onClick?: ((e: MouseEvent) => any) | null;
}

const MeasurementCard: React.FC<PropsType> = ({
  icon,
  title,
  value,
  unit,
  notificationLevel,
  decimal,
}: PropsType) => {
  const isValidValue = value !== null && value !== undefined;
  const roundedValue = value ? value.toFixed(decimal ? decimal : 0) : 0;

  return (
    <CardContainer style={{ color: colorMap[notificationLevel] }}>
      <IconContainer color={colorMap[notificationLevel]}>{icon}</IconContainer>
      <Title>{title}</Title>
      <MeasurementValueContainer>
        <MeasurementValue>
          {isValidValue ? roundedValue : 'N/A'}
        </MeasurementValue>
        {isValidValue && <Unit>{unit}</Unit>}
      </MeasurementValueContainer>
    </CardContainer>
  );
};

const CardContainer = styled.div`
  width: 100%;
  display: flex;
  align-items: center;

  @media only screen and (min-width: ${ScreenSize.xs}px) {
    border-bottom: 1px solid ${Colors.hb4};
    padding-bottom: 16px;
  }
`;

const IconContainer = styled.div`
  width: 30px;
  text-align: center;
  path {
    fill: ${(props) => props.color};
  }
`;

const Title = styled.div`
  font-family: Poppins;
  font-size: 16px;
  font-weight: 500;
  margin-left: ${Sizes.xLarge}px;
`;

const MeasurementValue = styled.div`
  font-family: Poppins;
  font-size: 30px;
  font-weight: 500;
  line-height: 40px;
`;

const MeasurementValueContainer = styled.div`
  display: flex;
  justify-content: center;
  margin-left: auto;
`;

const Unit = styled.span`
  font-family: Poppins;
  font-size: 14px;
  font-weight: 400;
  line-height: 13px;
  word-break: break-all;
  max-width: 40px;
  line-height: 18px;
  margin-left: 2px;
  text-align: right;
`;

export default MeasurementCard;
