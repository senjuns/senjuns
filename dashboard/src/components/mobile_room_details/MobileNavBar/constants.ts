export const RoomDetailTab = {
  Overview: 0,
  HeatMap: 1,
  LineChart: 2,
  PhotoFeed: 3,
};
