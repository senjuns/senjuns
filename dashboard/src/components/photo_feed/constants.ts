/**
 * TODO(NL-1391): improve the logic to remove the hardcode
 * These constants regarding platform specific need to be updated/removed sometime later in NL-1391.
 *
 * When the app is rendered on the mobile users, there are some headers and footers which need to be displayed.
 * And 280 seems to be the correct size for the mobile platform in order to display the photo feed in the whole device.
 */
export const MOBILE_PLATFORM_EXTRA_HEIGHT = 180;
/**
 * The height and width are just placeholders for non-mobile app users and we need to match this with the figma.
 */
export const NON_MOBILE_PLATFORM_PRESENTATION_HEIGHT = 400;
export const NON_MOBILE_PLATFORM_PRESENTATION_WIDTH = 400;

export const ZONE_MAP_URL = '/zone_map.png';
/**
 * The mini map height to be rendered on the screen.
 */
export const MINI_MAP_HEIGHT = 40;

/**
 * The zoom level limitations of the photo feed.
 */
export const ZOOM_LEVELS = {
  MAX: 2,
  NORMAL: 1,
  MIN: 0.5,
};

/**
 * When the section is unvisited, there needs to be opacity applied on it.
 */
export const UNVISITED_SECTION_OPACITY = 0.3;
