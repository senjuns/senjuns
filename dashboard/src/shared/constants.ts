import {
  MeasurementTypes,
  MeasurementStatisticsTypes,
  TPasswordRule,
} from '../shared/interfaces';

export const MeasurementUnit = {
  CelciusDegree: '°C',
  FahrenheitDegree: '°F',
  Percent: '%',
  KP: 'kP',
  mph: 'mph',
  ppm: 'ppm',
  DLI: 'DLI',
  μMol_m2_s: 'μMol/m2/s',
};

export const NotificationLevel = {
  Nominal: 'nominal',
  Warning: 'warning',
  Error: 'error',
};

export const ScreenSize = {
  xs: 576,
  sm: 768,
  md: 992,
  lg: 1200,
  xl: 1400,
};

export const Colors = {
  brightBlue100: '#60C3E3',
  brightBlue80: '#7FCEE8',
  brightBlue60: '#9FDAEE',
  grey100: '#E5E5E5',
  grey80: '#EAEAEA',
  grey60: '#EFEFEF',
  black: '#000000',
  dark1: '#383838',
  dark7: '#808080',
  dark8: '#B4B4B4',
  dark9: '#CFCFCF',
  green1: '#56C254',
  grey5: '#E0E0E0',
  hb1: '#82879A',
  hb2: '#9BA0B6',
  hb3: '#C3C7D9',
  hb4: '#ECEDF1',
  hb5: '#F0F1F4',
  hb6: '#F8F8FC',
  orange1: '#DD382F',
  orange2: '#E86339',
  orange3: '#F48E42',
  orange4: '#FFB94C',
  white: '#ffffff',
  linearGradientedOrange: 'linear-gradient(90deg, #E86339 0%, #FFB94C 100%)',
};

export const Sizes = {
  xxSmall: 4,
  xSmall: 8,
  small: 12,
  medium: 16,
  large: 20,
  xLarge: 24,
  xxLarge: 32,
};

export const Fonts = {
  h1: { size: 48, style: 'normal', weight: 600 },
  h2: { size: 40, style: 'normal', weight: 600 },
  h3: { size: 36, style: 'normal', weight: 600 },
  h4: { size: 24, style: 'normal', weight: 600 },
  h5: { size: 20, style: 'normal', weight: 600 },
  h6: { size: 18, style: 'normal', weight: 600 },
  body1: { size: 16, style: 'normal', weight: 'normal' },
  body2: { size: 14, style: 'normal', weight: 'normal' },
};

export const MEASUREMENT_OPTION_DETAILS = {
  [MeasurementTypes.Light]: {
    statisticsKey: MeasurementStatisticsTypes.AirTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
  [MeasurementTypes.Temperature]: {
    statisticsKey: MeasurementStatisticsTypes.AirTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
  [MeasurementTypes.Humidity]: {
    statisticsKey: MeasurementStatisticsTypes.RelativeHumidity,
    unit: MeasurementUnit.Percent,
  },
  [MeasurementTypes.LeafVpd]: {
    statisticsKey: MeasurementStatisticsTypes.LeafVpd,
    unit: MeasurementUnit.KP,
  },
  [MeasurementTypes.Vpd]: {
    statisticsKey: MeasurementStatisticsTypes.AirVpd,
    unit: MeasurementUnit.KP,
  },
  [MeasurementTypes.Airflow]: {
    statisticsKey: MeasurementStatisticsTypes.AirFlow,
    unit: MeasurementUnit.mph,
  },
  [MeasurementTypes.Co2]: {
    statisticsKey: MeasurementStatisticsTypes.AirCo2,
    unit: MeasurementUnit.ppm,
  },
  [MeasurementTypes.Ppfd]: {
    statisticsKey: MeasurementStatisticsTypes.Par,
    unit: MeasurementUnit.μMol_m2_s,
  },
  [MeasurementTypes.LeafTemperature]: {
    statisticsKey: MeasurementStatisticsTypes.LeafTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
};

export const APP_URL = {
  login: '/login',
  forgot: '/forgot',
  reset: '/reset',
  notFound: '/not-found',
  resetPassword: (username: string) => `/reset?username=${username}`,
  userHome: '/user/home',
  userAbout: '/user/about',
  home: '/',
  roomDetails: (id: number) => `/room-details/${id}`,
  roomDetailsOverview: (id: number) => `/room-details/${id}/overview`,
  roomDetailsHeatMap: (id: number) => `/room-details/${id}/heat-map`,
  roomDetailsLineChart: (id: number) => `/room-details/${id}/line-chart`,
  roomDetailsPhotoFeed: (id: number) => `/room-details/${id}/photo-feed`,
};

export enum ROOM_DETAIL_TABS {
  OVERVIEW = 'overview-tab',
  LINE_CHART = 'line-chart-tab',
  HEAT_MAP = 'heat-map-tab',
  PHOTO_FEED = 'photo-feed-tab',
}

export const AWS_API_CODE = {
  // will return when username or password incorrect or when user does not exists.
  user_not_found_exception: 'UserNotFoundException',
};

export const PASSWORD_RULES: TPasswordRule[] = [
  {
    label: 'Password must match',
    check: (password: string, confirmPassword: string): boolean =>
      password === confirmPassword,
  },
  {
    label: 'At least 8 characters',
    check: (password: string): boolean => password.length >= 8,
  },
  {
    label: 'Must contain at least one upper case letter',
    check: (password: string): boolean => /[A-Z]/.test(password),
  },
  {
    label: 'Must contain at least one special character',
    check: (password: string): boolean => /[#?!@$%^&*-]/.test(password),
  },
  {
    label: 'Must contain at least one number',
    check: (password: string): boolean => /[0-9]/.test(password),
  },
];

export const FEATURE_ENABLED = 'FEATURE_ENABLED';

export const NULL_VALUE = -1;

export const Z_INDEXES = {
  DROPDOWN: 10001,
};
