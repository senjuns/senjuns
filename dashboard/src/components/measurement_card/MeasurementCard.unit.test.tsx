import * as ShallowRenderer from 'react-test-renderer/shallow';
import MeasurementCard from './MeasurementCard';
import { ReactComponent as HumidityIcon } from 'assets/svg/humidity-icon.svg';
import { MeasurementUnit, NotificationLevel } from 'shared/constants';

describe('<MeasurementCard />', () => {
  it('renders correctly.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <MeasurementCard
        icon={<HumidityIcon />}
        title={'Humidity'}
        value={30}
        unit={MeasurementUnit.Percent}
        notificationLevel={NotificationLevel.Warning}
      />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
