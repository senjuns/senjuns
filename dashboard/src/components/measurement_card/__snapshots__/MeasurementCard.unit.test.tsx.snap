// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeasurementCard /> renders correctly. 1`] = `
<styled.div
  style={
    Object {
      "color": "#F48E42",
    }
  }
>
  <styled.div
    color="#F48E42"
  >
    <ForwardRef(SvgHumidityIcon) />
  </styled.div>
  <styled.div>
    Humidity
  </styled.div>
  <styled.div>
    <styled.div>
      30
    </styled.div>
    <styled.span>
      %
    </styled.span>
  </styled.div>
</styled.div>
`;
